tensorflow/core/platform/env_test.cc

```diff
@@ -340,4 +341,18 @@ TEST_F(DefaultEnvTest, LocalTempFilename) {
   EXPECT_FALSE(env->FileExists(filename).ok());
 }
 
+TEST_F(DefaultEnvTest, CreateNewFileWithUniqueName) {
+  Env* env = Env::Default();
+
+  string prefix = "tempfile-prefix-";
+  string suffix = ".tmp";
+  string filename = prefix;
+
+  EXPECT_TRUE(env->CreateNewFileWithUniqueName(&filename, suffix));
```

Comment:
Since we're now creating the file, should we delete it too?  Especially since we're not putting it in /tmp.

(Sorry I missed this before.  This used to be fine when we were just returning a file*name*.)

Reply:
Both `CreateNewFileWithUniqueName` and `LocalTempFilename` do not create new file, they just create new file name, check if it exists in file system. If does not exist, return `true`, otherwise clear string and return `false`.

`CreateNewFileWithUniqueName` -> `CreateUniqueFileName`?

Reply:
Ugh, sorry.  You're right, and that name sounds good.

Reply:
Done